//Row Sum
/*Problem Description
You are given a 2D matrix A of integers.
Your task is to compute the sum of elements in each row and return a 1D array where each element represents the sum of a corresponding row in the matrix.

Problem Constraints:
1 <= A.size() <= 103
1 <= A[i].size() <= 103
1 <= A[i][j] <= 103

Input Format:
First argument A is a 2D array of integers.(2D matrix).

Output Format:
Return an array containing row-wise sums of original matrix.

Example Input:
Input 1:
[1,2,3,4]
[5,6,7,8]
[9,2,3,4]

Example Output:
Output 1:
[10,26,18]

Example Explanation:
Explanation 1
Row 1 = 1+2+3+4 = 10
Row 2 = 5+6+7+8 = 26
Row 3 = 9+2+3+4 = 18
*/
package ArrayMatrix;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class RowSum {
    public static void main(String[] args) {
        List<List<Integer>> A = new ArrayList<>();
        A.add(Arrays.asList(1, 2, 3));
        A.add(Arrays.asList(4, 5, 6));
        A.add(Arrays.asList(7, 8, 9));

        List<Integer> ans = solve(A);
        for (Integer i : ans) {
            System.out.print(i + " ");
        }
    }

    public static List<Integer> solve(List<List<Integer>> A)
    {
        List<Integer> ans = new ArrayList<>();
        int sum = 0;
        for (int i = 0; i < A.size(); i++)
        {
            sum = 0;
            for (int j = 0; j < A.get(0).size(); j++)
            {
                sum += A.get(i).get(j);
            }
            ans.add(sum);
        }
        return ans;
    }
}
